src/update.js

```diff
@@ -0,0 +1,69 @@
+let glob = require('glob')
+let chalk = require('chalk')
+let series = require('run-series')
+let path = require('path')
+let child = require('child_process')
+let shared = require('./shared')
+
+/**
+ * updates dependencies to newer versions
+*/
+module.exports = function update(callback) {
+
+  // eslint-disable-next-line
+  let pattern = 'src/**/@(package\.json|requirements\.txt|Gemfile)'
+
+  let files = glob.sync(pattern).filter(function filter(filePath) {
+    if (filePath.includes('node_modules'))
+      return false
+    if (filePath.includes('vendor/bundle'))
+      return false
+    return true
+  })
+
+  series(files.map(file=> {
+    let cwd = path.dirname(file)
+    let options = {cwd}
+    return function updation(callback) {
+
+      // printer function
+      function exec(cmd, opts, callback) {
+        console.log(chalk.green(cwd))
+        console.log(chalk.bold.green(cmd))
+        child.exec(cmd, opts, callback)
+      }
+
+      // also a printer function
+      function done(err, stdout, stderr) {
+        if (err) {
+          console.log(chalk.bgRed.bold.white(err.message))
+          console.log(chalk.grey(err.stack))
+        }
+        if (stdout && stdout.length > 0) {
+          console.log(chalk.grey(stdout))
+        }
+        if (stderr && stderr.length > 0) {
+          console.log(chalk.yellow(stderr))
+        }
+        callback()
+      }
+
+      // TODO: I think we should consider what minimum version of node/npm this
+      // module needs to use as the npm commands below have different behaviour
+      // depending on npm versio - and enshrine those in the package.json
```

Comment:
Def need eyes on this as `npm update`'s behaviour has changed quite a bit from version to version, and I think that should impact what version of node we run against on CI.